include/gunrock/algorithms/hits.hxx

```diff
@@ -0,0 +1,375 @@
+/**
+ * @file hits.hxx
+ * @author Liyidong
+ * @brief Hyperlink-Induced Topic Search.
+ * @version 0.1
+ * @date 2021.05.06
+ *
+ * @copyright Copyright (c) 2020
+ *
+ */
+#pragma once
+
+#include <gunrock/algorithms/algorithms.hxx>
+
+#include <thrust/device_vector.h>
+#include <thrust/sequence.h>
+#include <thrust/copy.h>
+#include <thrust/execution_policy.h>
+
+#include <fstream>
+
+namespace gunrock {
+namespace hits {
+
+const int default_max_iterations = 50;
+
+class param_c {
+private:
+  int max_iterations;
+public:
+  param_c(int max_iterations) : max_iterations(max_iterations) {}
+  int get_max_iterations(){return this->max_iterations;}
+};// end of param_c
+
+template<typename graph_t>
+class result_c{
+     using vertex_t = typename graph_t::vertex_type;
+
+private:
+  int max_pages;
+  graph_t& G;
+
+  thrust::device_vector<float> auth;
+  thrust::device_vector<float> hub;
+  thrust::device_vector<vertex_t> auth_vertex;
+  thrust::device_vector<vertex_t> hub_vertex;
+
+public:
+  result_c(graph_t& G):G(G){}
+
+  void rank_authority(){
+    this->auth_vertex.resize(this->auth.size());
+    thrust::sequence(thrust::device,
+                     this->auth_vertex.begin(),
+                     this->auth_vertex.end(),
+                     0);
+    thrust::stable_sort_by_key(thrust::device,
+                               this->auth.begin(),
+                               this->auth.end(),
+                               this->auth_vertex.begin());
+  }
+
+  void rank_hub(){
+    this->hub_vertex.resize(this->hub.size());
+    thrust::sequence(thrust::device,
+                     this->hub_vertex.begin(),
+                     this->hub_vertex.end(),
+                     0);
+    thrust::stable_sort_by_key(thrust::device,
+                               this->hub.begin(),
+                               this->hub.end(),
+                               this->hub_vertex.begin());
+  }
+
+  void print_result(std::ostream& os = std::cout){
+    os<<"===Authority\n\n";
+    for(int i = 0; i < this->auth.size(); i++){
+      os<<"vertex ID: "<<this->auth_vertex[i]<<std::endl;
+      os<<"authority: "<<this->auth[i]<<std::endl;
+    }
+    os<<"===Hub\n\n";
+    for(int i = 0; i < this->hub.size(); i++){
+      os<<"vertex ID: "<<this->hub_vertex[i]<<std::endl;
+      os<<"authority: "<<this->hub[i]<<std::endl;
+    }
+  }
+
+  void print_result(int max_vertices, std::ostream& os = std::cout){
+    int vertices = (max_vertices < this->hub.size())
+              ? max_vertices
+              : this->hub.size();
+    os<<"===Authority\n\n";
+    for(int i = 0; i < vertices; i++){
+      os<<"vertex ID: "<<this->auth_vertex[i]<<std::endl;
+      os<<"authority: "<<this->auth[i]<<std::endl;
+    }
+    os<<"===Hub\n\n";
+    for(int i = 0; i < vertices; i++){
+      os<<"vertex ID: "<<this->hub_vertex[i]<<std::endl;
+      os<<"authority: "<<this->hub[i]<<std::endl;
+    }
+  }
```

Comment:
Can you use essential's print utility instead of writing a specialized helper here? It is clustering the file.